gateways/authorizenet/request_builders.go

```diff
@@ -109,3 +114,76 @@ func authentication(merchantName string, transactionKey string) MerchantAuthenti
 		TransactionKey: transactionKey,
 	}
 }
+
+func addL2L3Data(authRequest *sleet.AuthorizationRequest, authNetAuthRequest *CreateTransactionRequest) *CreateTransactionRequest {
+	if authRequest.Level3Data != nil {
+		authNetAuthRequest.TransactionRequest.LineItem = json.RawMessage(buildLineItemsString(authRequest))
+
+		authNetAuthRequest.TransactionRequest.Tax = &Tax{
+			Amount:  strconv.FormatInt(authRequest.Level3Data.TaxAmount.Amount, 10),
+		}
+
+		authNetAuthRequest.TransactionRequest.Duty = &Tax{
+			Amount:  strconv.FormatInt(authRequest.Level3Data.DutyAmount.Amount, 10),
+		}
+
+		authNetAuthRequest.TransactionRequest.Shipping = &Tax{
+			Amount:  strconv.FormatInt(authRequest.Level3Data.ShippingAmount.Amount, 10),
+		}
+
+		authNetAuthRequest.TransactionRequest.Customer = &Customer{
+			Id: authRequest.Level3Data.CustomerReference,
+		}
+	}
+
+	if authRequest.ShippingAddress != nil {
+		authNetAuthRequest.TransactionRequest.ShippingAddress = &BillingAddress{
+			FirstName: 	authRequest.CreditCard.FirstName,
+			LastName: 	authRequest.CreditCard.LastName,
+			Company: 	common.SafeStr(authRequest.ShippingAddress.Company),
+			Address: 	authRequest.ShippingAddress.StreetAddress1,
+			City:      	authRequest.ShippingAddress.Locality,
+			State:    	authRequest.ShippingAddress.RegionCode,
+			Zip:       	authRequest.ShippingAddress.PostalCode,
+			Country:   	authRequest.ShippingAddress.CountryCode,
+		}
+	}
+
+	return authNetAuthRequest
+}
+
+func buildLineItemsString(authRequest *sleet.AuthorizationRequest) string {
+	// No more than 30 lineitems
+	lineItemCount := len(authRequest.Level3Data.LineItems)
+	if lineItemCount > 30 {
+		lineItemCount = 30
+	}
+
+	lineItems := "{"
+	for i := 0; i < lineItemCount; i++ {
```

Comment:
```suggestion
	for i, lineItem := range authRequest.Level3Data.LineItems {
```
If we do this and then break when `i == 30` at the top of the for loop I feel like the logic is a bit more straightforward.